cfgov/v1/__init__.py

```diff
@@ -101,3 +102,25 @@ def get_protected_url(context, page):
             return url.replace(page_hostname, staging_hostname)
         else:
             return '#'
+
+@contextfunction
+def related_metadata_tags(context, page):
+    request = context['request']
+    # Set the tags to correct data format
+    tags = {'links': []}
+    # From an ancestor, get the form ids then use the first id since the 
+    # filterable list on the page will probably have the first id on the page.
+    id = None
+    filter_page = None
+    for ancestor in page.get_ancestors().reverse().specific():
+        if ancestor.specific_class.__name__ in ['BrowseFilterablePage', 'SublandingFilterablePage']:
+            filter_page = ancestor
+            id = util.get_form_id(ancestor, request.GET)
+            break
```

Comment:
It's reversed. It's looking for a (Browse/Sublanding)FilterablePage, then if it finds one it will prepend the URL to the link back.